src/web/source.rs

```diff
@@ -41,24 +41,24 @@ impl ToJson for FileList {
 
         m.insert("metadata".to_string(), self.metadata.to_json());
 
-        let mut file_vec: Vec<Json> = Vec::new();
-
-        for file in &self.files {
-            let mut file_m: BTreeMap<String, Json> = BTreeMap::new();
-            file_m.insert("name".to_string(), file.name.to_json());
-
-            file_m.insert(match file.file_type {
-                              FileType::Dir => "file_type_dir".to_string(),
-                              FileType::Text => "file_type_text".to_string(),
-                              FileType::Binary => "file_type_binary".to_string(),
-                              FileType::RustSource => "file_type_rust_source".to_string(),
-                          },
-                          true.to_json());
-
-            file_vec.push(file_m.to_json());
-        }
-
-        m.insert("files".to_string(), file_vec.to_json());
+        let files = self.files.iter()
+            .map(|file| {
+                let mut file_map: BTreeMap<String, Json> = BTreeMap::new();
```

Comment:
Can you do the same trick from `temp_json` above where you use the same temporary allocation for each file?

Reply:
I'm not sure it'll have as much of an effect due to how BTrees work, but I'll certainly do it